File: .claude/worktrees/exciting-newton-e27414/backend/routers/notifications.py
from fastapi import APIRouter, Depends
from sqlalchemy import text
from sqlalchemy.ext.asyncio import AsyncConnection

from database import get_db
from auth import get_current_user

router = APIRouter(prefix="/api/notifications", tags=["notifications"])


@router.get("")
async def list_notifications(
    current_user: dict = Depends(get_current_user),
    db: AsyncConnection = Depends(get_db),
):
    rows = (await db.execute(
        text("SELECT * FROM notifications WHERE user_id = :uid ORDER BY created_at DESC LIMIT 30"),
        {"uid": current_user["id"]},
    )).mappings().all()
    return [dict(r) for r in rows]


@router.patch("/read-all")
async def mark_all_read(
    current_user: dict = Depends(get_current_user),
    db: AsyncConnection = Depends(get_db),
):
    await db.execute(
        text("UPDATE notifications SET `read` = 1 WHERE user_id = :uid"),
        {"uid": current_user["id"]},
    )
    await db.commit()
    return {"ok": True}


@router.delete("/clear-all", status_code=204)
async def clear_all_notifications(
    current_user: dict = Depends(get_current_user),
    db: AsyncConnection = Depends(get_db),
):
    await db.execute(
        text("DELETE FROM notifications WHERE user_id = :uid"),
        {"uid": current_user["id"]},
    )
    await db.commit()


@router.patch("/{notif_id}/read")
async def mark_read(
    notif_id: int,
    current_user: dict = Depends(get_current_user),
    db: AsyncConnection = Depends(get_db),
):
    await db.execute(
        text("UPDATE notifications SET `read` = 1 WHERE id = :id AND user_id = :uid"),
        {"id": notif_id, "uid": current_user["id"]},
    )
    await db.commit()
    return {"ok": True}

File: .claude/worktrees/nice-bartik-75f638/api/routes/notifications.php
<?php
// Routes: /api/notifications/*
require_once __DIR__ . '/../db.php';
require_once __DIR__ . '/../auth_middleware.php';
require_once __DIR__ . '/../helpers.php';

$db  = get_db();
$id  = isset($segments[1]) && is_numeric($segments[1]) ? (int)$segments[1] : null;
$sub = $segments[1] ?? ''; // could be 'read-all' or 'clear-all'

// GET /api/notifications
if ($method === 'GET' && $sub === '') {
    $u    = auth_user();
    $stmt = $db->prepare('SELECT * FROM notifications WHERE user_id=? ORDER BY created_at DESC LIMIT 30');
    $stmt->execute([(int)$u['id']]);
    $rows = $stmt->fetchAll();
    foreach ($rows as &$r) { $r['id'] = (int)$r['id']; $r['read'] = (int)$r['read']; }
    respond($rows);
}

// PATCH /api/notifications/read-all
elseif ($method === 'PATCH' && $sub === 'read-all') {
    $u = auth_user();
    $db->prepare('UPDATE notifications SET `read`=1 WHERE user_id=?')->execute([(int)$u['id']]);
    respond(['ok' => true]);
}

// DELETE /api/notifications/clear-all
elseif ($method === 'DELETE' && $sub === 'clear-all') {
    $u = auth_user();
    $db->prepare('DELETE FROM notifications WHERE user_id=?')->execute([(int)$u['id']]);
    http_response_code(204);
    exit;
}

// PATCH /api/notifications/{id}/read
elseif ($method === 'PATCH' && $id !== null && ($segments[2] ?? '') === 'read') {
    $u = auth_user();
    $db->prepare('UPDATE notifications SET `read`=1 WHERE id=? AND user_id=?')->execute([$id, (int)$u['id']]);
    respond(['ok' => true]);
}

else {
    respond(['detail' => 'Not found'], 404);
}
